### PR TITLE
Create a Deaths lookup

### PR DESCRIPTION
**Pull request by Moohan**
This is now done, I've split out the changes into other PRs which can/should be reviewed first to make things simpler!
 
- [x] #698 
- [x] #699 
- [x] Fix R CMD build warnings

### CODE REVIEW

File: R/run_episode_file.R [hunk 1]
```diff
@@ -326,3 +326,20 @@ join_cohort_lookups <- function(ep_file_data, year) {
...
+  return(
+    ep_file_data %>%
+      dplyr::left_join(
+        slf_deaths_lookup,
+        by = "chi",
+        na_matches = "never",
+        relationship = "many-to-one"
+      )
+  )
+}
```
Comment (Moohan) [inline R/run_episode_file.R:345]: Put this in it's own script so it can also be used in the indiv file

Comment (Jennit07) [reply R/run_episode_file.R:345]: Done

File: R/process_lookup_deaths.R [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#'
+#' @description Currently this just uses the NRS death dates 'as is', with no
+#' corrections or modifications, it is expected that this will be expanded to
+#' use the CHI deaths extract from IT as well as taking into account data in
+#' the episode file to assess the validity of a death date.
+#'
+#' @param year
+#' @param nrs_deaths_data_path
+#' @param chi_deaths_data_path
+#' @param write_to_disk
```
Comment (Jennit07) [inline R/process_lookup_deaths.R:11]: Update documentation descriptions

Comment (Jennit07) [reply R/process_lookup_deaths.R:11]: Done

**Comment by Jennit07** [comment]
I've pushed some changes to fix the CMD build warnings. If you are happy with the code please let me know and i will review again

File: R/link_delayed_discharge_eps.R [hunk 3]
```diff
@@ -50,7 +50,7 @@ link_delayed_discharge_eps <- function(data, year) {
     )
 
   by_dd <- dplyr::join_by(
-    chi,
+    "chi",
```
Comment (Moohan) [inline R/link_delayed_discharge_eps.R:53]: Does this definitely work?

I broke it previously by doing something very similar 😅

**Comment by Moohan** [comment]
> I've pushed some changes to fix the CMD build warnings. If you are happy with the code please let me know and I will review it again

I'm happy if you are 😄 I checked the R CMD fixes and they looked good so I just cherry-picked them to main-R

**Review by Jennit07 — APPROVED** [review]
We have both reviewed this and it looks good. Think it is ready to merge!

**Pull request merged**